newTestFile.py

```diff
@@ -0,0 +1,92 @@
+try:
+    import pyaudio
+    import numpy as np
+    import pylab
+    import matplotlib.pyplot as plt
+    import time
+    import sys
+except:
+    print ("Something didn't import")
+
+i=0
+f,ax = plt.subplots(2)
+global counter, tempArray

```

Comment:
remove globals. Very rare to need globals also. Pass through as arguments instead. Let me know if there's a global you dont think is removable, and I'll see if i can refactor it